client/signup/main.jsx

```diff
@@ -237,26 +245,37 @@ class Signup extends Component {
 	}
 
 	componentDidUpdate( prevProps ) {
+		const { flowName, stepName, signupDependencies, sitePlanName, sitePlanSlug } = this.props;
+
 		if (
-			( this.props.flowName !== prevProps.flowName ||
-				this.props.stepName !== prevProps.stepName ) &&
+			( flowName !== prevProps.flowName || stepName !== prevProps.stepName ) &&
 			! this.state.shouldShowLoadingScreen
 		) {
-			recordSignupStep( this.props.flowName, this.props.stepName, this.getRecordProps() );
+			recordSignupStep( flowName, stepName, this.getRecordProps() );
 		}
 
 		if (
-			get( this.props.signupDependencies, 'siteType' ) !==
-			get( prevProps.signupDependencies, 'siteType' )
+			get( signupDependencies, 'siteType' ) !== get( prevProps.signupDependencies, 'siteType' )
 		) {
 			this.startTrackingForBusinessSite();
 		}
 
-		if ( this.props.stepName !== prevProps.stepName ) {
+		if ( stepName !== prevProps.stepName ) {
 			this.preloadNextStep();
 			// `scrollToTop` here handles cases where the viewport may fall slightly below the top of the page when the next step is rendered
 			this.scrollToTop();
 		}
+
+		if ( sitePlanSlug !== prevProps.sitePlanSlug ) {
+			recordSignupPlanChange(
```

Comment:
This is firing with incorrect values when a plan is originally selected.

When I selected the personal plan for a new site, I got `calypso_signup_plan_change: {previousPlanSlug: null, currentPlanSlug: 'free_plan' }` and  then another event that seemed to reverse this `calypso_signup_plan_change: {previousPlanSlug: 'free_plan', currentPlanSlug: null }`.

With a personal plan, I selected a premium theme and selected to upgrade plan, this logged the event as expected 👍 

It looks like `sitePlanName` and `sitePlanSlug` don't represent the value of the plan that the user selects the first time on the `/start/plans` page.